### ISSUE TITLE
TRG-Check Pipeline Issues

### ISSUE THREAD

**Issue by ds-mwesener** [eclipse-tractusx/sig-infra#341]
**Is your support request related to a problem? Please describe.**  
Failing trg-checks in pipeline.

[Pipeline Run](https://github.com/catenax-ng/tx-traceability-foss/actions/runs/6795774976/job/18474501962)

- [Docker File which fails](https://github.com/catenax-ng/tx-traceability-foss/blob/chore/TRACEFOSS-XXX-fix-trg/Dockerfile)
- [Helm Chart which is unable to render](https://github.com/catenax-ng/tx-traceability-foss/blob/chore/TRACEFOSS-XXX-fix-trg/charts/traceability-foss/Chart.yaml)

**Describe the solution you'd like**  
Due to a correct configuration I expect the pipeline to be successful.

**Additional context**  
Already talked to @SebastianBezold regarding my issues.

**Comment by SebastianBezold** [comment]
As i already suspected, our current implementation of the non root user check does not resolve the variable references used in `USER  ${UID}:${GID}`. It's currently seen as invalid user. I'll see, if we can improve the check to also resolve that and then have it as valid user.

The Dockerfile is following the TRGs correctly. The automation itself is the issue here.

**Comment by SebastianBezold** [comment]
The Chart rendering fails with following error: 
`"traceability-foss/charts/frontend/templates/deployment.yaml" at <.Values.imagePullSecret.dockerconfigjson>: nil pointer evaluating interface {}.dockerconfigjson`

Running `helm template` on the Chart directly works as expected. I'll check if there is a know issue or newer version of the helm library we use

**Comment by tomaszbarwicki** [comment]
Yes, it is a known, reported issue: https://github.com/helm/helm/issues/12385

**Comment by SebastianBezold** [comment]
Ah cool thanks for pointing out. Didn't know that it is even reported by you 😁

**Comment by tomaszbarwicki** [comment]
Hi @ds-mwesener, apologies for late response, it seems the latest fix to quality check tool covers your case. I tested it on the provided forked repo and on the original at tractus-x, both went through. Please try run it again and let us know the result.

cc: @FaGru3n

**Comment by almadigabor** [comment]
Hey @ds-mwesener! Does the trg check show correct results now? If yes, I would close this issue.

**Comment by FaGru3n** [comment]
If your fine with the result i would close this issue.

**Comment by ds-mwesener** [comment]
Hi @almadigabor / @FaGru3n we still have issues with the pipeline:
![image](https://github.com/eclipse-tractusx/sig-infra/assets/124587888/e7c5520f-6efd-40da-9d50-653e15bc34c8)

**Comment by hzierer** [comment]
moved to our backlog to plan it properly
